packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2672,11 +2672,9 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
   // On iOS, tapping the status bar scrolls the app's primary scrollable to the
   // top. We implement this by providing a primary scroll controller and
   // scrolling it to the top when tapped.
-
-  final ScrollController _primaryScrollController = ScrollController();
-
   void _handleStatusBarTap() {
-    if (_primaryScrollController.hasClients) {
+    final ScrollController? _primaryScrollController = PrimaryScrollController.of(context);
```

Comment:
The comment above probably needs updating (we are no longer providing the primary scroll controller here.

Reply:
Fixed!